pkg/controller/vitessshard/reconcile_disk.go

```diff
@@ -0,0 +1,141 @@
+package vitessshard
+
+import (
+	"context"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"vitess.io/vitess/go/vt/topo/topoproto"
+
+	planetscalev2 "planetscale.dev/vitess-operator/pkg/apis/planetscale/v2"
+	"planetscale.dev/vitess-operator/pkg/operator/results"
+	"planetscale.dev/vitess-operator/pkg/operator/rollout"
+)
+
+const (
+	pvcFilesystemResizeAnnotation = "planetscale.com/pvc-filesystem-resize"
+)
+
+func (r *ReconcileVitessShard) reconcileDisk(ctx context.Context, vts *planetscalev2.VitessShard) (reconcile.Result, error) {
+	resultBuilder := &results.Builder{}
+
+	// If we're already cascading a disk size update, we don't need to look further.
+	if rollout.Cascading(vts) {
+		return resultBuilder.Result()
+	}
+
+	anythingChanged := false
+	tabletPods, err := r.tabletPodsFromShard(ctx, vts)
+	if err != nil {
+		return resultBuilder.Error(err)
+	}
+
+	for i  := range vts.Spec.TabletPools {
+		tabletPool := &vts.Spec.TabletPools[i]
+		if tabletPool.DataVolumeClaimTemplate == nil {
+			continue
+		}
+
+		requestedDiskQuantity, ok := tabletPool.DataVolumeClaimTemplate.Resources.Requests[v1.ResourceStorage]
+		if !ok {
+			continue
+		}
+
+		poolTablets, err := tabletKeysForPool(vts, tabletPool.Cell, string(tabletPool.Type))
```

Comment:
Why not just have this function take the tablet pool type, rather than forcing the caller to use type casting?

Reply:
Yeah good point. Fixed it to take the tablet pool type.